app/assets/javascripts/backbone/views/note_view.coffee

```diff
@@ -0,0 +1,56 @@
+class MS.Views.NoteView extends Backbone.View
+
+  initialize: (params) ->
+    new MS.Views.AutoLoadingIndicatorView()
+    if @newRecord() then @editView() else @showView()
+
+  events:
+    'click a.edit-action': 'editView'
+    'click .cancel': 'showView'
+    'submit .note-form': 'update'
+    'ajax:success .delete-action': 'remove'
+
+  newRecord: ->
+    @$el.data('id') == 'new'
+
+  editView: (e) ->
+    e.preventDefault() if e
+    @$('.view-block').hide()
+    @$('.form-block').show()
+    @resizeTextareas()
+
+  showView: (e) ->
+    e.preventDefault() if e
+    if @newRecord()
+      @$el.remove()
+    else
+      @$('.view-block').show()
+      @$('.form-block').hide()
+
+  update: (e) ->
+    $form = @$(e.target).closest('form')
+    # We send form data via ajax so we can capture the response from server
+    $.post($form.attr('action'), $form.serialize())
+      .done (response) =>
+        @$('.view-block').html(response)
+        @$('.view-block').show()
+        @$('.form-block').hide()
+      .fail (response) =>
```

Comment:
Actually I just removed this because it was redundant with the main application view's error handler. I'm not sure which was taking precedence or how they might have interfered with each other.